tests/unit/test_charm.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Jenkins-k8s charm unit tests."""
+
+# pylint:disable=protected-access
+
+from functools import partial
+from typing import Any, Callable, cast
+from unittest.mock import MagicMock
+
+import pytest
+import requests
+from ops.charm import PebbleReadyEvent
+from ops.model import ActiveStatus, BlockedStatus, Container, StatusBase
+from ops.testing import Harness
+
+import jenkins as jenkins_src
+from charm import JenkinsK8SOperatorCharm
+from jenkins import LAST_EXEC, UPDATE_VERSION, unlock_jenkins
+
+from .helpers import make_relative_to_path
+from .types_ import ContainerWithPath
+
+
+def test_unlock_jenkins(
+    harness: Harness,
+    container_with_path: ContainerWithPath,
+    mocked_get_request: Callable[[str, int, Any, Any], requests.Response],
+    monkeypatch: pytest.MonkeyPatch,
+    jenkins_version: str,
+):
+    """
+    arrange: given a mocked container and a monkeypatched Jenkins client.
+    act: unlock_jenkins is called.
+    assert: files necessary to unlock Jenkins are written.
+    """
+    monkeypatch.setattr(requests, "get", partial(mocked_get_request, status_code=403))
+    harness.begin()
+
+    unlock_jenkins(container_with_path.container)
+
+    assert (
+        make_relative_to_path(container_with_path.path, LAST_EXEC).read_text(encoding="utf-8")
+        == jenkins_version
+    )
+    assert (
+        make_relative_to_path(container_with_path.path, UPDATE_VERSION).read_text(encoding="utf-8")
+        == jenkins_version
+    )
+
+
+def test__on_jenkins_pebble_ready_no_container(harness: Harness):
+    """
+    arrange: given a pebble ready event with container unable to connect.
+    act: when the Jenkins pebble ready event is fired.
+    assert: the event should be deferred.
+    """
+    harness.set_can_connect(harness.model.unit.containers["jenkins"], False)
+    harness.begin()
+    charm = cast(JenkinsK8SOperatorCharm, harness.charm)
+    mock_event = MagicMock(spec=PebbleReadyEvent)
+    mock_event.workload = None
+
+    charm._on_jenkins_pebble_ready(mock_event)
+
+    mock_event.defer.assert_called()
+
+
+@pytest.mark.parametrize(
+    "status_code,expected_status",
+    [
+        pytest.param(503, BlockedStatus, id="jenkins not ready"),
+        pytest.param(200, ActiveStatus, id="jenkins ready"),
+    ],
+)
+def test__on_jenkins_pebble_ready(  # pylint: disable=too-many-arguments
```

Comment:
Should this, and any other disables, have an explanatory comment on them?

Reply:
fixed, thanks!